backend/src/Edna/Config/Environment.hs

```diff
@@ -0,0 +1,104 @@
+module Edna.Config.Environment
+  ( EdnaEnvironments(..)
+
+  , ednaEnvs
+  , askDebugDB
+  ) where
+
+import Universum
+
+import Control.Exception (throwIO)
+import Data.ByteString.Char8 (pack)
+import qualified Data.Char as C
+import Edna.Config.Definition (LoggingConfig(..))
+import Edna.Util (ConnString(..), NetworkAddress)
+import System.Environment (lookupEnv)
+import qualified Text.Show
+
+data EnvParseError = WrongNetworkAddress
+                   | WrongMaxConnections
+
+instance Show EnvParseError where
+  show WrongNetworkAddress = "Can't parse EDNA_API_LISTEN"
+  show WrongMaxConnections = "Can't parse EDNA_DB_MAX_CONNECTIONS"
+
+instance Exception EnvParseError
+
+readEnv :: Read a => String -> EnvParseError -> IO (Maybe a)
+readEnv name err = do
+  env <- lookupEnv name
+  case readEither <$> env of
+    Nothing -> return Nothing
+    Just (Right value) -> return $ Just value
+    Just (Left _) -> throwIO err
+
+getBoolEnv :: String -> IO (Maybe Bool)
+getBoolEnv name = lookupEnv name <&> \case
+  Nothing                        -> Nothing
+  Just "1"                       -> Just True
+  Just (map C.toLower -> "true") -> Just True
+  _                              -> Just False
+
+-- If @EDNA_DB_DEBUG@ environment variable is set to @1@
+-- or @TRUE@ (case-insensitive), we will run postgres with
+-- debug logging.
+askDebugDB :: IO Bool
+askDebugDB = do
+  env <- getBoolEnv "EDNA_DB_DEBUG"
+  return $ fromMaybe False env
+
+data EdnaEnvironments = EdnaEnvironments
+  { eeListen :: IO (Maybe NetworkAddress)
+  , eeDocs :: IO (Maybe Bool)
+  , eeConnection :: IO (Maybe ConnString)
+  , eeMaxCon :: IO (Maybe Int)
+  , eeLogging :: IO (Maybe LoggingConfig)
+  } deriving stock (Generic)
+
+ednaEnvs :: EdnaEnvironments
+ednaEnvs = EdnaEnvironments
+  { eeListen = apiListenEnv
+  , eeDocs = serveDocsEnv
+  , eeConnection = connectionEnv
+  , eeMaxCon = dbMaxConnectionsEnv
+  , eeLogging = loggingLevelEnv
+  }
+
+apiListenEnv :: IO (Maybe NetworkAddress)
+apiListenEnv = readEnv "EDNA_API_LISTEN" WrongNetworkAddress
+
+serveDocsEnv :: IO (Maybe Bool)
+serveDocsEnv = getBoolEnv "EDNA_SERVE_DOCS"
+
+connectionEnv :: IO (Maybe ConnString)
+connectionEnv = do
+  host <- lookupEnv "POSTGRES_HOST"
+  port <- lookupEnv "POSTGRES_PORT"
+  user <- lookupEnv "POSTGRES_USER"
+  password <- lookupEnv "POSTGRES_PASSWORD"
+  db <- lookupEnv "POSTGRES_DB"
+
+  if any isNothing [host, port, user, password, db]
```

Comment:
I couldn't find it in docs, but AFAIU if one keyword appears in `conn-string`, the last one takes precedence. So for example the following works fine: `"host=1.2.3.4 host=localhost dbname=ednadb user=ednauser password=ednapass"`. For this reason I think it's not necessary to require all variables to be defined. For example, if only `POSTGRES_HOST` is defined, we can append `host=$POSTGRES_HOST` to the connection string and it should work fine (override the host if it was provided in the config).

Reply:
Yes, you are right. I've tried `"host=1.2.3.4 port=5432 dbname=test user=ednauser password=ednapass host=localhost dbname=ednadb"` and it's working.